plugins/ipam/host-local/backend/disk/backend.go

```diff
@@ -108,13 +110,36 @@ func (s *Store) ReleaseByID(id string) error {
 		if err != nil {
 			return nil
 		}
-		if strings.TrimSpace(string(data)) == strings.TrimSpace(id) {
+		if strings.TrimSpace(string(data)) == (strings.TrimSpace(id) + "\n" + ifname) {
 			if err := os.Remove(path); err != nil {
 				return nil
 			}
+			found = true
 		}
 		return nil
 	})
+
+	if (!found) && (err == nil) {
+		err = filepath.Walk(s.dataDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
Another approach would be to pull the directory Walk/data compare/file remove code out to a new function, and then just call it twice; once for id+\n+iface, and if that wasn't found a second time for just 'id'.